### PR TITLE
Use upstream SCF tiling in CPU codegen

### PR DESCRIPTION
**Pull request by pzread**
Use the upstream SCF tiling to replace `LinalgTilingPattern`.

This change also removes the `peeled-loops` option from `LinalgSingleTilingExpert`, which is unused and unsupported in the upstream (and we have a separate pass for peeling).

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp [hunk 1]
```diff
@@ -58,19 +58,42 @@ static FailureOr<Operation *> getRootOp(func::FuncOp funcOp) {
   return rootOp;
 }
 
+/// Builds a proper tile sizes vector for the op.
+/// scf::tileUsingSCFForOp expects the num of tile sizes = num of loops. This
+/// method returns a proper tile sizes vector for each op during tiling.
+static SmallVector<Value> buildTileSizesForOp(OpBuilder &b, Operation *op,
+                                              SmallVector<int64_t> tileSizes) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp:66]: I think you can cast this to `TilingInterface` and assert. You need to only check that the op implements the tiling interface. Also I dont think you need to change the insertion point (but there is no harm in it).

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Transforms.cpp [hunk 2]
```diff
@@ -459,16 +491,20 @@ struct LinalgStrategyTilePass
...
-      tilingPattern.add<LinalgTilingPattern>(anchorOpName, ctx, options,
-                                             filter);
+      tilingPattern.add<LinalgSCFTilingPattern>(anchorOpName, ctx, options,
+                                                filter);
     else
-      tilingPattern.add<LinalgTilingPattern>(ctx, options, filter);
-    if (anchorOpName == tensor::PadOp::getOperationName())
-      populatePadTensorTilingPatterns(tilingPattern, options);
+      tilingPattern.add<LinalgSCFTilingPattern>(ctx, options, filter);
+    if (anchorOpName == tensor::PadOp::getOperationName()) {
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Transforms.cpp:498]: actually the upstream tiling handles pad ops as well. So you should be able to drop line 498 - 502.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
THis is awesome! Just a few nits. 
Also I was under the impression that there was some machinations w.r.t lowering config and use of `scf::tileUsingSCFForOp` and `scf::tileConsumerAndFuseProducersGreedilyUsingSCFForOp`. I dont see it in the PR. Was this not an issue, or was it resolved.

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Transforms.cpp (hunk 2 above)
Comment (pzread) [reply llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Transforms.cpp]: Done.

File: compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp (hunk 1 above)
Comment (pzread) [reply compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp]: Done. Using the current insertion point but still restore it with the guard.

**Comment by pzread** [comment]
> THis is awesome! Just a few nits. Also I was under the impression that there was some machinations w.r.t lowering config and use of `scf::tileUsingSCFForOp` and `scf::tileConsumerAndFuseProducersGreedilyUsingSCFForOp`. I dont see it in the PR. Was this not an issue, or was it resolved.

Sorry for the confusion (I know this likely will happen 😅 ). This PR only migrates the tiling, which is required for the migration of tile-and-fuse #10770.

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**